editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/MethodSelectionDialog.java

```diff
@@ -0,0 +1,236 @@
+	private String message = ""; //$NON-NLS-1$
+	private String title = ""; //$NON-NLS-1$
+
+	public MethodSelectionDialog(Shell parent, ILabelProvider renderer) {
+		super(parent);
+		labelProvider = renderer;
+	}
+
+	public MethodSelectionDialog(Shell parent) {
+		this(parent, new JavaElementLabelProvider(JavaElementLabelProvider.SHOW_DEFAULT));
+	}
+
+	@Override
+	protected Control createDialogArea(Composite parent) {
+		super.setTitle(title);
+		super.setMessage(message);
+
+		Composite area = new Composite(parent, SWT.NULL);
+		GridLayout gridLayout = new GridLayout(1, false);
+		area.setLayout(gridLayout);
+		area.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+		// list of elements
+		final FilteredTree elementTable = new FilteredTree(area, SWT.SINGLE | SWT.FULL_SELECTION | SWT.BORDER,
+				new LocalPatternFilter(), true);
+		elementTable.getFilterControl().addFocusListener(new FocusListener() {
+            @Override
+            public void focusGained(FocusEvent e) {
+                // ignore
+            }
+
+            @Override
+            public void focusLost(FocusEvent e) {
+                getButton(IDialogConstants.OK_ID).setEnabled(validate());
+            }
+        });
+		elementTable.getFilterControl().addKeyListener(new KeyListener() {
+            @Override
+            public void keyPressed(KeyEvent e) {
+                // ignore
+            }
+
+            @Override
+            public void keyReleased(KeyEvent e) {
+                getButton(IDialogConstants.OK_ID).setEnabled(validate());
+            }
+        });
+		
+		treeViewer = elementTable.getViewer();
+		treeViewer.addPostSelectionChangedListener(input -> 
+			getButton(IDialogConstants.OK_ID).setEnabled(validate())
+		);
+		treeViewer.getControl().setLayoutData(new GridData(GridData.FILL_BOTH));
+		treeViewer.setContentProvider(new MethodTreeContentProvider());
+		treeViewer.setLabelProvider(labelProvider);
+		treeViewer.setInput(elementArray);
+
+		if (elementArray == null || elementArray.length == 0) {
+			super.setMessage(UIMessages.beanConfigUtilNoMethodsAvailable,  IMessageProvider.ERROR);
+			treeViewer.getTree().setEnabled(false);
+			elementTable.getFilterControl().setEnabled(false);
+		}
+
+		return area;
+	}
+
+	/**
+	 * Set the list of available elements to choose from.
+	 * @param elements
+	 */
+	public void setElements(Object[] elements) {
+		elementArray = elements;
+	}
+
+	/**
+	 * Returns an array of the currently selected elements. To be called within or
+	 * after open().
+	 * 
+	 * @return returns an array of the currently selected elements.
+	 */
+	protected Object[] getSelectedElements() {
+		Assert.isNotNull(treeViewer);
+		IStructuredSelection ssel = treeViewer.getStructuredSelection();
+		return ssel.toArray();
+	}
+
+	/**
+	 * Returns the first element from the list of results. Returns <code>null</code>
+	 * if no element has been selected.
+	 *
+	 * @return object or null
+	 */
+	public Object getFirstResult() {
```

Comment:
You could decide to use Selections.getFirstSelection(...) instead of rewriting that code here.